### PR TITLE
Project: Test Your Code

### PR DESCRIPTION
**Pull request by kubilaycaglayan**
Dear TSE,

1. We used the existing repository and created an isolated branch and branched out from the **develop** branch.
2. Initialized RSpec.
3. Added **main_spec.rb** file in the **spec** folder.
4. **Added specs** for all public methods we have previously created.
5. Wrapped specs for each method in describe block.
6. **Used let**.
7. Used **one it block for each expected behavior**.
8. Tested **positive and negative scenarios**.   .to, .not_to 

Here are our tested classes and public methods:

### class Game 💎 

1. :initialize, 
2. :board, 
3. :names, 
4. :greeting

### class TicTacToe < Game 💎 

1. :initialize, 
2. :board, 
3. :greeting, 
4. :to_s, 
5. :check_game, 
6. :check_spot, 
7. :names, 
8. :win_message

### class Player 💎 

1. :talk_with, 
2. :name

### And two methods from main.rb file 💎 

1. :turn_sign,
2. :check_name

_Note: These two methods are the only methods from the main.rb file that doesn't involve User Interactions._

With all our respects...

### CODE REVIEW

File: spec/main_spec.rb [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+describe TicTacToe do
+  let(:x_wins) { [1, 1, 1, 0, 0, 0, 0, 0, 0] }
+  let(:y_wins) { [10, 10, 10, 0, 0, 0, 0, 0, 0] }
+  let(:draw) { [1, 1, 10, 10, 10, 1, 1, 10, 1] }
+  start_game = TicTacToe.new('Ikraam', 'Kubilay')
+  describe '#initialize' do
+    it 'Returns an ' do
+      expect { TicTacToe.new('Ikraam', 'Kubilay') }.not_to raise_error
+    end
+  end
```
Comment (euqueme) [inline spec/main_spec.rb:38]: - [x]  The description for the initialize it block 'Returns an' looks unfinished add a little more to it explaining what does it return.

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @kblycaglayan and @ikraamg :balloon: :lollipop: 

### You've done a great job with your test cases it's good practice to separate each class and class method in describe blocks and have it blocks for each expected behavior and you've achieved this beautifully so claps to you :clap: :clap: :clap: 

### There are still some aspects to work on :construction: :construction_worker: :construction_worker: so please keep reading :point_down: 

* [x] specs have to cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) )

- I see all your test cases are positive so let's add negative test cases at least for all the test cases that validate boolean results for example:

  in your describe '#check_game' this 'it' block could be changed to be a negative test case like this:

```ruby
#Positive 
    it 'returns false if the game is goin on' do
      start_game.board = [1, 10, 0, 0, 0, 0, 0, 0, 0]
      expect(start_game.check_game).to be false
    end

#Negative
    it "doesn't return true if the game is goin on" do
      start_game.board = [1, 10, 0, 0, 0, 0, 0, 0, 0]
      expect(start_game.check_game).not_to be true
    end
```
* [x]  [`let` use](https://github.com/rubocop-hq/rspec-style-guide#let-blocks). ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/before_each.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

- I see you've used let in some of your test cases but you could still use it for the variables in your Game and Player test cases

### That's it, team! :raised_hands: 

### Please :pray: do the changes :man_technologist: :man_technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :technologist: 

PROJECT STATUS: REQUESTED CHANGES :traffic_light:

**Comment by kubilaycaglayan** [comment]
Dear TSE,

We applied the necessary changes.
**1. Negative test case added
2. Let variables added for Game and Player test cases
3. initialize description corrected**
Thank you very much for your efforts and time.

With all our respect...

**Review by mosesogwo — APPROVED** [review]
**HI team:clap:. Good work, well written PR description:+1:. Your project is approved :heavy_check_mark:**

You can now merge your pull request.

**Cheers :stars:**

### Status: Approved!

**Pull request merged**